### PR TITLE
Update the DG with current implementation of `TA Assist`

### PR DESCRIPTION
**Pull request by njxue**
Partially completes #82 
- Renamed all instances of `AddressBook` to `TA Assist` in DG
- Updated (and removed) diagrams to match current implementation
   * I'm not sure if `ParserStudentIndexUtil` should be part of the ParserClasses diagram
- Added sequence diagrams for `assign` and `focus` command
  * Not sure if `hasModuleClass` should be omitted from the focus command diagram. If it's included inside, we might need an opt or alt to indicate that `enterFocusMode` only happens if `hasModuleClass` is true which complicates the diagram
- Added description for implementation of the focus state/command
  * I used `unfocus` instead of `back` since I think that's what we're changing to

### CODE REVIEW

**Review by Bubbl3T — APPROVED** [review]
LGTM! There seems to already be some UML diagrams involving the sessions, we might need to update them as we update the session class. Thanks for doing the refactoring of AB3 to TA Assist too! Also, nice spotting that the `ArgumentMultimap` class depends on the `Prefix` class and adding it to the diagram. :)

File: docs/diagrams/StorageClassDiagram.puml [hunk 1]
```diff
@@ -29,15 +29,16 @@ HiddenOutside ..> Storage
...
 JsonUserPrefsStorage .up.|> UserPrefsStorage
-JsonAddressBookStorage .up.|> AddressBookStorage
-JsonAddressBookStorage ..> JsonSerializableAddressBook
-JsonSerializableAddressBook --> "*" JsonAdaptedStudent
-JsonAdaptedStudent --> "*" JsonAdaptedTag
+JsonTaAssistStorage .up.|> TaAssistStorage
+JsonTaAssistStorage ..> JsonSerializableTaAssist
+JsonSerializableTaAssist--> "*" JsonAdaptedStudent
+JsonSerializableTaAssist--> "*" JsonAdaptedModuleClass
+JsonAdaptedStudent --> "*" JsonAdaptedModuleClass
```
Comment (Berted) [inline docs/diagrams/StorageClassDiagram.puml]:
```suggestion
```

In PR #114, this is not the case. `JsonAdaptedStudent` actually stores a list of `String`-s of the names of the `ModuleClass`-es. Whilst not the most intuitive design, this allows the `json` file to be more compact with less redundant information.

**Review by Berted — CHANGES_REQUESTED** [review]
Nice work! Just a minor nit regarding the `Storage` class diagram, as it differs from the current implementation as of PR #114.

**Review by Berted — APPROVED** [review]
LGTM!

**Pull request merged**